### PR TITLE
Issue # 12932: Improve grammar in AbstractClassNameCheck

### PR DESCRIPTION
**Pull request by Bharath-Ganesh**
Issue #12932
Improve grammar in AbstractClassNameCheck

### CODE REVIEW

**Comment by nrmancuso** [comment]
@Bharath-Ganesh please make CI happy, looks like you referenced PR number instead of issue

**Comment by Bharath-Ganesh** [comment]
> @Bharath-Ganesh please make CI happy, looks like you referenced PR number instead of issue



> @Bharath-Ganesh please make CI happy, looks like you referenced PR number instead of issue

I'm afraid that I failed the pipeline again. 
I can't understand why it's failing. Could you please direct me ?

**Comment by nrmancuso** [comment]
@Bharath-Ganesh looks like failure in azure is random, not related to your changes. Y

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
  * <p>
  * Rationale: Abstract classes are convenience base class implementations of
  * interfaces, not types as such. As such they should be named to indicate this.
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java:36]: `not types as such` -> `and are not considered types per se.`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java [hunk 2]
```diff
@@ -34,7 +34,7 @@
...
- * Also, if names of classes starts with 'Abstract' it's very convenient that
+ * Also, if the names of classes start with 'Abstract' it's very convenient that
  * they will have abstract modifier.
```
Comment (nrmancuso) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java:38]: `As such they should be named to indicate this. Also, if the names of classes start with 'Abstract' it's very convenient that they will have abstract modifier.` -> `For this reason, it should be made obvious that a given class is abstract by prefacing the class name with 'Abstract'.`

**Review by nrmancuso — CHANGES_REQUESTED** [review]
Items:

**Comment by romani** [comment]
@Bharath-Ganesh, please reply with comment "done" to each review it, do not resolve them.

**Review by nrmancuso — APPROVED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java (hunk 1 above)
Comment (Bharath-Ganesh) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java (hunk 2 above)
Comment (Bharath-Ganesh) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java]: done

**Comment by Bharath-Ganesh** [comment]
> @Bharath-Ganesh, please reply with comment "done" to each review it, do not resolve them.

@romani Thx! Will make a note of it.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java [hunk 3]
```diff
@@ -33,9 +33,9 @@
  * </p>
  * <p>
  * Rationale: Abstract classes are convenience base class implementations of
- * interfaces, not types as such. As such they should be named to indicate this.
- * Also, if names of classes starts with 'Abstract' it's very convenient that
- * they will have abstract modifier.
+ * interfaces, and are not considered types per se. For this reason, it should
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java:36]: > considered types per se

I do not understand this this English.  please rephrase in simple way, that will be clearly understood by any non-English people.

you might just need to remove ", not types as such."

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java (hunk 3 above)
Comment (Bharath-Ganesh) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java:36]: How about the below one? 

Rationale: Abstract classes are convenience base class implementations of
interfaces. For this reason, it should be made obvious that a given class is 
abstract by prefacing the class name with 'Abstract'.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java:36]: Good

Comment (Bharath-Ganesh) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java]: done

**Comment by romani** [comment]
Fix CI, do not resolve review items - just reply "done"

**Comment by Bharath-Ganesh** [comment]
> Fix CI, do not resolve review items - just reply "done"

Could you help me figure out the reason for the CI failure? 
If this is not due to code change, can we trigger a CI build without making any commit?

**Comment by romani** [comment]
Click on details link follow until

**Comment by Bharath-Ganesh** [comment]
Looking at the error trace, I'm not able to identify the reason for the failure. 
Also, I have ensured all the `mvn clean verify` was working from local. 

image

Looking at the azure logs
`     - Method <com.puppycrawl.tools.checkstyle.checks.naming.ParameterNameCheck.mustCheckName(com.puppycrawl.tools.checkstyle.api.DetailAST)> calls method <com.puppycrawl.tools.checkstyle.api.DetailAST.getType()> in (ParameterNameCheck.java:235)\
     - Method <com.puppycrawl.tools.checkstyle.checks.naming.ParameterNameCheck.mustCheckName(com.puppycrawl.tools.checkstyle.api.DetailAST)> calls method <com.puppycrawl.tools.checkstyle.api.DetailAST.getParent()> in (ParameterNameCheck.java:238)\
     - Method <com.puppycrawl.tools.checkstyle.checks.naming.MethodNameCheck.visitToken(com.puppycrawl.tools.checkstyle.api.DetailAST)> calls method <com.puppycrawl.tools.checkstyle.api.DetailAST.findFirstToken(int)> in (MethodNameCheck.java:239)\
+    - Method <com.puppycrawl.tools.checkstyle.checks.naming.AbstractClassNameCheck.isAbstract(com.puppycrawl.tools.checkstyle.api.DetailAST)> calls method <com.puppycrawl.tools.checkstyle.api.DetailAST.findFirstToken(int)> in (AbstractClassNameCheck.java:241)\
     - Method <com.puppycrawl.tools.checkstyle.checks.naming.AbstractClassNameCheck.isAbstract(com.puppycrawl.tools.checkstyle.api.DetailAST)> calls method <com.puppycrawl.tools.checkstyle.api.DetailAST.findFirstToken(int)> in (AbstractClassNameCheck.java:242)\
     - Method <com.puppycrawl.tools.checkstyle.checks.naming.MethodNameCheck.visitToken(com.puppycrawl.tools.checkstyle.api.DetailAST)> calls method <com.puppycrawl.tools.checkstyle.api.DetailAST.getParent()> in (MethodNameCheck.java:242)\
-    - Method <com.puppycrawl.tools.checkstyle.checks.naming.AbstractClassNameCheck.isAbstract(com.puppycrawl.tools.checkstyle.api.DetailAST)> calls method <com.puppycrawl.tools.checkstyle.api.DetailAST.findFirstToken(int)> in (AbstractClassNameCheck.java:243)\
     - Method <com.puppycrawl.tools.checkstyle.checks.naming.MethodNameCheck.visitToken(com.puppycrawl.tools.checkstyle.api.DetailAST)> calls method <com.puppycrawl.tools.checkstyle.api.DetailAST.findFirstToken(int)> in (MethodNameCheck.java:244)\
     - Method <com.puppycrawl.tools.checkstyle.checks.naming.MethodNameCheck.visitToken(com.puppycrawl.tools.checkstyle.api.DetailAST)> calls method <com.puppycrawl.tools.checkstyle.api.DetailAST.getText()> in (MethodNameCheck.java:252)\
     - Method <com.puppycrawl.tools.checkstyle.checks.naming.MethodNameCheck.visitToken(com.puppycrawl.tools.checkstyle.api.DetailAST)> calls method <com.puppycrawl.tools.checkstyle.api.DetailAST.getText()> in (MethodNameCheck.java:253)\
 `
The  DetailAST.getText() is failing. Can you help me understand what might have caused this failure?

**Review by romani — APPROVED** [review]
ok to merge

**Comment by romani** [comment]
you need to apply patch to your code suggested at https://dev.azure.com/romanivanovjr/romanivanovjr/_build/results?buildId=13937&view=logs&j=c902ebb4-c9f8-5f09-4e17-ff78fbbc842e&t=9ca98c81-ff64-58f0-9d03-a23ac1c4a111&l=1245

unfortunately line numbers are shifted.

**Comment by Bharath-Ganesh** [comment]
Thanks, @romani .  I'm reading about the ArchUnit. https://www.archunit.org/userguide/html/000_Index.html
I'll revert to you.

**Review by romani — APPROVED** [review]

**Pull request merged**